lib/active_model/serializer/adapter.rb

```diff
@@ -1,7 +1,10 @@
+require 'active_model/serializer/cache'
+
 module ActiveModel
   class Serializer
     class Adapter
       extend ActiveSupport::Autoload
+      include ActiveModel::Serializer::Cache
```

Comment:
So, my suggestion was to really move the code out of the former class. Instead, you just moved it to a new file, but the code is still inside the Adapter.

In my opinion, the main problem with this approach is that you can't unit test the code you added, but only have integration tests, which aren't really good for refactoring the code later. My suggestion was to use composition, or in other words, call the cache class such as:

```ruby
def cached_object
   CachedResource.new(serializer).load
end
```

With that, you can unit test `CachedResource` separately. Besides that, the adapter has a huge amount of methods doing things that are not the adapter's responsibility.

What do you think? cc @guilleiguaran